enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/log/EntryRecord.java

```diff
@@ -64,7 +64,7 @@ public static EntryRecord read( ReadableChannel channel, ChannelMarshal<Replicat
         }
         catch ( ReadPastEndException e )
         {
-            throw EndOfStreamException.INSTANCE;
+            throw new EndOfStreamException( e );
```

Comment:
We were losing the underlying exception so it wasn't obvious which line threw the ReadPastEndException